### PR TITLE
emit close event if  client connection was lost

### PR DESCRIPTION
**Pull request by ipa-mdl**

### CODE REVIEW

**Comment by gonzalocasas** [comment]
@ipa-mdl thanks for the PR! How about moving this to the `Ros` client instead? Since `comm` is an implementation detail, it feels like the more appropriate way of triggering a `close` event would be at the level of the ROS client. Especially taking into account this other PR https://github.com/gramaziokohler/roslibpy/pull/3 that will add a second communication layer.

What do you think?

**Comment by ipa-mdl** [comment]
I have just put it to the same file that is emitting "ready" (which is named "connection" in roslibjs).
How to add this to `Ros`  if the callback is called in  `comm`?

**Comment by ipa-mdl** [comment]
> I have just put it to the same file that is emitting "ready" (which is named "connection" in roslibjs).

This is still the same in #3: https://github.com/gramaziokohler/roslibpy/blob/16936186f757a70673f23a4472a4690c17504217/src/roslibpy/comm/comm_autobahn.py#L80

**Comment by gonzalocasas** [comment]
Sorry, my comment was not very clear. What I meant to say is that there should be a way to use the `close` event from the `Ros` object, just as `ready` is available on `Ros` but actually triggered by the factory. If not, you will end up with client code that hooks into the factory event handlers, like `ros.factory.on('close', callback)` and that couples the client code too much to the implementation details of the comm layer.

To summarize, it's ok to leave the `emit` where it is, but I would add an `on_close` to the factory to hook into it, and a matching `on_close` on the `Ros` class that calls it on the factory. Does that sound good?

**Comment by ipa-mdl** [comment]
I have just used it like this:
```
r = Ros(url)
r.on('ready', my_ready_handler)
r.on('close', my_close_handler)
```

**Comment by gonzalocasas** [comment]
Ah, ok, you're right, we already have this generic forwarding in place. Then let's merge!
